Pushing a branch with x.y-z pattern in rally-tracks causes error when rally tries to determine version.

Creating a branch following an unusual pattern like `9.2-something-like-this` is followed by an error in determining the best_match version: 

```
$ esrally race --distribution-version="9.1.4" --track=geoshape --test-mode --on-error=abort --runtime-jdk=bundled
[..]
[ERROR] Cannot race. Traceback (most recent call last):
  File "rally/esrally/actor.py", line 95, in guard
    return f(self, msg, sender)
  File "rally/esrally/racecontrol.py", line 114, in receiveMsg_Setup
    self.coordinator.setup(sources=msg.sources)
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
  File "rally/esrally/racecontrol.py", line 228, in setup
    self.current_track = track.load_track(self.cfg, install_dependencies=True)
                         ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "rally/esrally/track/loader.py", line 217, in load_track
    repo = track_repo(cfg)
  File "rally/esrally/track/loader.py", line 334, in track_repo
    return GitTrackRepository(cfg, fetch, update)
  File "rally/esrally/track/loader.py", line 378, in __init__
    self.repo.update(distribution_version)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File "rally/esrally/utils/repo.py", line 63, in update
    branch = versions.best_match(git.branches(self.repo_dir, remote=self.remote), distribution_version)
  File "rally/esrally/utils/versions.py", line 180, in best_match
    if version_type == "with_minor" and (latest_minor := latest_bounded_minor(available_alternatives, versions)):
                                                         ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "rally/esrally/utils/versions.py", line 216, in latest_bounded_minor
    major, minor, patch, suffix = components(a, strict=False)
                                  ~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "rally/esrally/utils/versions.py", line 63, in components
    return int(matches.group(1)), int(matches.group(2)), int(matches.group(3)), matches.group(4)
                                                         ~~~^^^^^^^^^^^^^^^^^^
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
```

--- a/esrally/utils/versions.py
+++ b/esrally/utils/versions.py
@@ -22,7 +22,7 @@ from esrally import exceptions
 
 VERSIONS = re.compile(r"^(\d+)\.(\d+)\.(\d+)(?:-(.+))?$")
 
-VERSIONS_OPTIONAL = re.compile(r"^(\d+)(?:\.(\d+))?(?:\.(\d+))?(?:-(.+))?$")
+VERSIONS_OPTIONAL = re.compile(r"^(\d+)(?:\.(\d+)?(?:\.(\d+)?(?:-(.+))?)?)?$")
 
 
 def _versions_pattern(strict):

--- a/tests/utils/versions_test.py
+++ b/tests/utils/versions_test.py
@@ -44,6 +44,8 @@ class TestsVersions:
         assert versions.is_version_identifier("5", strict=False)
         assert versions.is_version_identifier("23", strict=False)
         assert versions.is_version_identifier("20.3.7-SNAPSHOT", strict=False)
+        assert versions.is_version_identifier("9.2-test", strict=False) is False
+        assert versions.is_version_identifier("9-test", strict=False) is False
 
     def test_is_serverless(self):
         assert versions.is_serverless("serverless")

